<img style="float: left; margin: 30px 15px 15px 15px;" src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTFzQj91sOlkeDFkg5HDbjtR4QJYmLXkfMNig&usqp=CAU" width="400" height="500" /> 
    
    
### <font color='navy'> Simulación matemática. 

**Nombres:** Francisco Javier Balcazar Orrantia. Maximiliano Jakez Salazar

**Fecha:** 14 de noviembre del 2022.

**Expediente** : 737642.
    
**Profesor:** Oscar David Jaramillo Zuluaga.
    
**Link Github**: Link con el enlace del repositorio del creador del proyecto

# Tarea 8: Clase 18

## Tarea.
> - Consultar tasas de interés reales en algún banco y proyectar un ahorro mensual para que al terminar su carrera tengan $\$85000$ MXN en su cuenta.
> - Hacer esto en un nuevo archivo, llamarlo Gi_T7ApellidoNombre.ipynb y subirlo a canvas. Donde $i=1,2$ es el número del grupo al que pertenece 
> - Plazo: 

### Ejercicio 1: (enunciado) Transcribirlo

Código de solución estudiante 1

In [13]:
# Código de solución estudiante 1
# Librería de análisis de datos
import pandas as pd
# Librería para manipulación de fechas
from datetime import date
import numpy as np
import numpy_financial as npf

In [20]:
# Datos para la cuenta de ahorro
tasa_interes = 0.105
years = 2
Abono_year = 12
inicial = 68000
meta = 85000
fecha_inicio = (date(2022,11,14))

In [21]:
# Cálculo de abonos mensuales iguales para llegar a la meta de ahorro en el tiempo deseado
Abono = -npf.pmt(tasa_interes/Abono_year,years*Abono_year,(meta-inicial))
Abono

788.3927078693152

In [22]:
# Rango de fechas en la tabla
rng = pd.date_range(fecha_inicio, periods=years*Abono_year, freq='M')
rng.name = 'Fecha del abono'
# Columnas de abonos y extracto de cuenta en las fechas respectivas
df = pd.DataFrame(index = rng,columns=['Abono','Balance'],dtype='float')
# Para poner el indice con los periodos y no con las fechas
df.reset_index(inplace=True)

df.index+=1
df.index.name='Periodo'
df.head()

Fecha del abono  Abono  Balance
Periodo                                
1            2022-11-30    NaN      NaN
2            2022-12-31    NaN      NaN
3            2023-01-31    NaN      NaN
4            2023-02-28    NaN      NaN
5            2023-03-31    NaN      NaN

In [23]:
npf.ppmt(tasa_interes/Abono_year, 1, years * Abono_year, Abono)

-29.66409685446769

In [24]:
# Los abonos son iguales
df['Abono'] = Abono
# Vector indizador de los periodos de los abonos
index_vector = np.arange(1,len(rng)+1)
# Extracto de la cuenta mes a mes
df["Balance"] = inicial * (1 + tasa_interes/Abono_year)**index_vector + \
                Abono * (((1 + tasa_interes/Abono_year)**index_vector - 1)/(tasa_interes/Abono_year))

df.head()

Fecha del abono       Abono       Balance
Periodo                                          
1            2022-11-30  788.392708  69383.392708
2            2022-12-31  788.392708  70778.890102
3            2023-01-31  788.392708  72186.598098
4            2023-02-28  788.392708  73606.623539
5            2023-03-31  788.392708  75039.074203

In [25]:
# Redondear con dos cifras decimales y mostrar DataFrame
df = df.round(2)
df

Fecha del abono   Abono    Balance
Periodo                                   
1            2022-11-30  788.39   69383.39
2            2022-12-31  788.39   70778.89
3            2023-01-31  788.39   72186.60
4            2023-02-28  788.39   73606.62
5            2023-03-31  788.39   75039.07
6            2023-04-30  788.39   76484.06
7            2023-05-31  788.39   77941.69
8            2023-06-30  788.39   79412.07
9            2023-07-31  788.39   80895.32
10           2023-08-31  788.39   82391.54
11           2023-09-30  788.39   83900.86
12           2023-10-31  788.39   85423.39
13           2023-11-30  788.39   86959.24
14           2023-12-31  788.39   88508.52
15           2024-01-31  788.39   90071.36
16           2024-02-29  788.39   91647.88
17           2024-03-31  788.39   93238.19
18           2024-04-30  788.39   94842.42
19           2024-05-31  788.39   96460.68
20           2024-06-30  788.39   98093.11
21           2024-07-31  788.39   99739.81
22           2024-08-31  788.39  101400.93
23           2024-09-30  788.39  103076.58
24           2024-10-31  788.39  104766.89

### Respuesta planteada en el ejercicio usando (markdown)
 La respuesta es .........$68,000 
 
 y abonando a cada mes $788
 
 Banco Compartamos, tasa de interes=10.5%

Código de solución estudiante 2

In [27]:
# Código de solución estudiante 2
# Librería de análisis de datos
import pandas as pd
# Librería para manipulación de fechas
from datetime import date
import numpy as np
import numpy_financial as npf

In [28]:
# Datos para la cuenta de ahorro
tasa_interes = 0.105
years = 2
Abono_year = 1
inicial = 70000
meta = 85000
fecha_inicio = (date(2022,11,14))

In [34]:
# Cálculo de abonos mensuales iguales para llegar a la meta de ahorro en el tiempo deseado
Abono = -npf.pmt(tasa_interes/Abono_year,years*Abono_year,(meta-inicial))
Abono

8700.89073634204

In [33]:
# Rango de fechas en la tabla
rng = pd.date_range(fecha_inicio, periods=years*Abono_year, freq='M')
rng.name = 'Fecha del abono'
# Columnas de abonos y extracto de cuenta en las fechas respectivas
df = pd.DataFrame(index = rng,columns=['Abono','Balance'],dtype='float')
# Para poner el indice con los periodos y no con las fechas
df.reset_index(inplace=True)

df.index+=1
df.index.name='Periodo'
df.head()

Fecha del abono  Abono  Balance
Periodo                                
1            2022-11-30    NaN      NaN
2            2022-12-31    NaN      NaN

In [5]:
npf.ppmt(tasa_interes/Abono_year, 1, years * Abono_year, Abono)

NameError: name 'npf' is not defined

In [35]:
# Los abonos son iguales
df['Abono'] = Abono
# Vector indizador de los periodos de los abonos
index_vector = np.arange(1,len(rng)+1)
# Extracto de la cuenta mes a mes
df["Balance"] = inicial * (1 + tasa_interes/Abono_year)**index_vector + \
                Abono * (((1 + tasa_interes/Abono_year)**index_vector - 1)/(tasa_interes/Abono_year))

df.head()

Fecha del abono        Abono        Balance
Periodo                                            
1            2022-11-30  8700.890736   86050.890736
2            2022-12-31  8700.890736  103787.125000

In [36]:
# Redondear con dos cifras decimales y mostrar DataFrame
df = df.round(2)
df

Fecha del abono    Abono    Balance
Periodo                                    
1            2022-11-30  8700.89   86050.89
2            2022-12-31  8700.89  103787.12

### Respuesta planteada en el ejercicio usando (markdown)
 La respuesta es .........$70,000
 
 Banco Compartamos, tasa de interes=10.5%